Analytics/Classes/Internal/SEGSegmentIntegration.m

```diff
@@ -109,17 +109,44 @@ - (id)initWithAnalytics:(SEGAnalytics *)analytics httpClient:(SEGHTTPClient *)ht
         if ([NSThread isMainThread]) {
             [self setupFlushTimer];
         } else {
-            dispatch_sync(dispatch_get_main_queue(), ^{
+            dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
This one is interesting. After updating Xcode I started getting random deadlocks when testing, even on a clean repo on `origin/master` without any changes and `DerivedData` nuked.

The debugger showed that two threads (`main` and our serial queue) are locking the main queue : 
<img width="1148" alt="screen shot 2018-07-13 at 14 13 34" src="https://user-images.githubusercontent.com/5746414/42696287-0b9de3e0-86b8-11e8-9b1a-de097da99aec.png">
<img width="1148" alt="screen shot 2018-07-13 at 14 13 47" src="https://user-images.githubusercontent.com/5746414/42696288-0bc78d8a-86b8-11e8-98dc-b87e2c7a726e.png">

Because we just need to set the timer to let ARC keep a reference on it I replaced the code to use `async`  instead of `sync`, which should prevent any deadlock on this section. What do you think @f2prateek?

Reply:
I actually think we've gone back and forth on this. I can try digging up the history, you might be able to find it via git blame as well.